src/main/java/Task.java

```diff
@@ -0,0 +1,32 @@
+public class Task {
```

Comment:
Perhaps this `Task` class and it's derived class could be put into a single package?